[DOCS] 8.3.0 Release Notes (#2030)
elastic/security-docs: docs/release-notes/8.3.asciidoc

[[release-notes-header-8.3.0]]
== 8.3

[discrete]
[[release-notes-8.3.0]]
== 8.3.0

[discrete]
[[breaking-changes-8.3.0]]
==== Breaking changes
// tag::breaking-changes[]
// NOTE: The breaking-changes tagged regions are reused in the Elastic Installation and Upgrade Guide. The pull attribute is defined within this snippet so it properly resolves in the output.
:pull: https://github.com/elastic/kibana/pull/

> * Adds a confirmation modal with link to documentation when updating rules if V1/V2 jobs are installed ({pull}128334[#128334]).

Reviewer: ```suggestion
* Updates Elastic prebuilt {ml} detection rules for some Windows and Linux anomalies with new `v3` {ml} jobs. A confirmation modal is displayed when updating rules if `v1`/`v2` jobs are installed. If you're using 8.2 or earlier versions of {beats} or {agent}, you may need to duplicate prebuilt rules or create new custom rules _before_ you update the prebuilt rules. Once you update the prebuilt rules, they will only use `v3` {ml} jobs. Refer to <<ml-job-compatibility>> for more information.({pull}128334[#128334]).
```

// end::breaking-changes[]

[discrete]
[[features-8.3.0]]
==== Features
* Renames Endpoint Security integration to "{endpoint-cloud-sec}" ({pull}132752[#132752]).
* Adds a “Cases” widget that shows cases by status to the Detection & Response page ({pull}130670[#130670]).
* The Detection & Response dashboard provides focused visibility into the day-to-day operations of your security environment. It helps security operations managers and analysts quickly monitor recent and high priority detection alerts and cases, and identify the hosts and users associated with alerts ({pull}128335[#128335], {pull}129021[#129021], {pull}128087[#128087], {pull}131828[#131828], {pull}131029[#131029]).
* Adds a *User risk* tab to the User details flyout ({pull}130256[#130256]).
* Adds an *Authentications* tab to the User details flyout ({pull}129456[#129456]).
* Adds the ability to investigate Osquery results in Timeline ({pull}128596[#128596]).
* Allows multiple alerts to be added to a case ({pull}130958[#130958]).
* Adds the option to delete case comments from a case ({pull}130254[#130254]).
* Provides an option to select a severity level for a case ({pull}131626[#131626]).

> * Adds the *Alerts* tab to cases ({pull}131883[#131883]).

Reviewer 2: I think is better to say that you can view all alerts attached to a case so the users can understand better what this feature is about. Also, the feature is experimental. Maybe it is worth mentioning that also. What do you think?

Author: Ah, I see what you're saying about being more descriptive @cnasikas. How about:
Adds the *Alerts* tab to cases, which is an experimental feature that allows users to inspect attached alerts.

Reviewer 2: Perfect!

** `add_tags`
** `delete_tags`
** `set_tags`
** `add_index_patterns`
** `delete_index_patterns`
** `set_index_patterns`
** `set_timeline`
* Fixes a bug that caused the rule details page to crash when users opened a deleted or non-existent rule ({pull}133867[#133867]).
* Allows threshold alerts to be investigated in Timeline if filters are not provided ({pull}133733[#133733]).
* Prevents events from being added to cases from Timeline ({pull}133410[#133410]).
* Fixes a bug that prevented the Users and Hosts pages from resetting after being sorted ({pull}133111[#133111]).
* Removes the filter and investigate in Timeline options from the {agent} status in highlighted fields ({pull}132829[#132829], {pull}132586[#132586]).
* Improves the copy of Timeline tooltips ({pull}132756[#132756]).
* Fixes a validation bug that occurred when users were building a rule exception and changed the exception statement’s operator ({pull}131989[#131989]).
* Adds a checkmark to the pagination selection on the *Exceptions lists* page ({pull}131979[#131979]).
* Re-adds the success message that displays when users export an exceptions list ({pull}131952[#131952]).

> * Updates import toast logic ({pull}131873[#131873]).

Author: Hm, I'm not sure why this line isn't getting updated when I commit edits/push local edits up. Anyways, I'll commit this change and look into why later.

[[release-notes-header-8.3.0]]
== 8.3

[discrete]
[[release-notes-8.3.0]]
== 8.3.0

[discrete]
[[breaking-changes-8.3.0]]
==== Breaking changes
// tag::breaking-changes[]
// NOTE: The breaking-changes tagged regions are reused in the Elastic Installation and Upgrade Guide. The pull attribute is defined within this snippet so it properly resolves in the output.
:pull: https://github.com/elastic/kibana/pull/

> * Adds confirmation modal with docs link when updating rules if V1/V2 jobs are installed ({pull}128334[#128334]).

Reviewer: I'm still working on a good description of this feature and the upgrade/breaking considerations, which I may need to sync with you separately from this PR. For now, a few small tweaks:
```suggestion
* Adds confirmation modal with docs link when updating prebuilt {ml} rules if V1/V2 {ml} jobs are installed ({pull}128334[#128334]).
```

// end::breaking-changes[]

[discrete]
[[features-8.3.0]]
==== Features
* Enables additional fields for prebuilt detection rules: `related_integrations`, `required_fields`, and `setup` ({pull}132409[#132409]).

> * Adds a “Related integrations” column to the rules table, adds a “Setup” tab to the Rule details flyout, and adds “Related integrations” and “Required fields” sections to the Rule details flyout’s Definition section ({pull}131475[#131475]).

Author: I'll remove this summary for now since the latest consensus is that this feature isn't being included in 8.3.

* Adds a “Cases” widget that shows cases by status to the Detection & Response page ({pull}130670[#130670]).
* Adds “Vulnerable hosts by severity” and “Vulnerable users by severity” tables to the Detection & Response page ({pull}128335[#128335]).
* Adds an “Open alerts by rule” table that shows up to four rules sorted by severity and then by alert count to the Detection & Response page ({pull}129021[#129021]).
* Adds an “Alerts” widget to the Detection & Response page with donut charts that show alerts by status and severity ({pull}128087[#128087]).
* Enhances Detection & Response page widgets by adding pagination and making numbers you click open directly in timeline ({pull}131828[#131828]).
* Adds a “Recently created cases” widget to the Cases page ({pull}131029[#131029]).
* Adds a “User risk” tab to the User details flyout ({pull}130256[#130256]).

> * Adds an “Authentications” tab to the User details flyout ({pull}129456[#129456]).

Reviewer: 👍 Good idea, thanks!

* Adds the ability to investigate Osquery results in Timeline ({pull}128596[#128596]).
* Allows multiple alerts to be added to a case ({pull}130958[#130958]).
* Adds the option to delete case comments multiple alerts to be added to a case ({pull}130254[#130254]).
* Provides an option to select a severity level for a case ({pull}131626[#131626]).
* Adds the *Alerts* tab to cases ({pull}131883[#131883]).
* Adds the *Average time to close* metric to the Cases page ({pull}131909[#131909]).

[discrete]
[[bug-fixes-8.3.0]]
==== Bug fixes and enhancements
* Separates array values in the Alerts table with commas ({pull}133297[#133297]).
* Exposes the EQL search settings `event_category_field`, `tiebreaker_field`, and `timestamp_field` through the rules API and UI for Event Correlation rules ({pull}132247[#132247]).
* Adds the *Session ID* field to the *Highlighted fields* section of the Alert details flyout ({pull}132219[#132219]).

> * Introduces a new design for the main navigation menu ({pull}132210[#132210], {pull}131437[#131437]).

Author: @machadoum and @semd if you agree that we should move this summary up to the new features section, we can take care of that now - thanks!

> * Introduces a new design for the main navigation menu ({pull}132210[#132210], {pull}131437[#131437]).

Reviewer 3: I agree with moving the summary to the features section!

> * Introduces a new design for the main navigation menu ({pull}132210[#132210], {pull}131437[#131437]).

Author: Moved to new features, thanks!

> * Adds Dashboards and Threat Hunting Landing pages ({pull}130905[#130905]).

Author: Think this is actually part of the new nav design, so it should be lumped in with the summary for #131437. @machadoum and @semd would you mind confirming that I'm correct, or clarifying if I'm not.

> * Adds Dashboards and Threat Hunting Landing pages ({pull}130905[#130905]).

Reviewer 3: Yes, The landing pages are part of the menu. You can omit it.
But I think that Dashboards are a reference to this pages:
<img width="231" alt="Screenshot 2022-06-28 at 10 20 54" src="https://user-images.githubusercontent.com/1490444/176130634-f74dfe4f-84fb-435c-8aa5-d1aff7eaebc0.png">

> * Adds Dashboards and Threat Hunting Landing pages ({pull}130905[#130905]).

Author: @machadoum ah, so then should the summary for this PR be more about the new Dashboards section being created/added to the main nav?

> * Adds Dashboards and Threat Hunting Landing pages ({pull}130905[#130905]).

Author: Chatted with @semd in Slack about adding https://github.com/elastic/kibana/pull/133719 to the summary for the new nav design.

** `add_tags`
** `delete_tags`
** `set_tags`
** `add_index_patterns`
** `delete_index_patterns`
** `set_index_patterns`
** `set_timeline`
* Fixes a bug that caused the rule details page to crash when users opened a deleted or non-existent rule ({pull}133867[#133867]).
* Allows threshold alerts to be investigated in Timeline if filters are not provided ({pull}133733[#133733]).
* Prevents events from being added to cases from Timeline ({pull}133410[#133410]).
* Fixes All Users and All Hosts page should reset the current page when sorting ({pull}133111[#133111]).
* Removes the filter and investigate in Timeline options from the Elastic Agent status in Highlighted fields ({pull}132829[#132829], {pull}132586[#132586]).
* Improves the copy of Timeline tooltips ({pull}132756[#132756]).
* Fixes a validation bug that occurred when users were building a rule exception and changed the exception statement’s operator ({pull}131989[#131989]).
* Adds a checkmark to the pagination selection on the *Exceptions lists* page ({pull}131979[#131979]).
* Re-adds the success message that displays when users export an exceptions list ({pull}131952[#131952]).

> * Updates import toast logic ({pull}131873[#131873]).

Reviewer: Pretty unclear -- maybe a few more words could help:
```suggestion
* Updates notification logic for importing rules ({pull}131873[#131873]).
```

Author: Thanks, @joepeeples ! Was saving this till you merged https://github.com/elastic/security-docs/pull/2084.

* Adds the *Average time to close* metric to the Cases page ({pull}131909[#131909]).

[discrete]
[[bug-fixes-8.3.0]]
==== Bug fixes and enhancements
* Separates array values in the Alerts table with commas ({pull}133297[#133297]).
* Exposes the EQL search configurations `event_category_field`, `tiebreaker_field`, and `timestamp_field` through the rules API and UI for Event Correlation rules ({pull}132247[#132247]).
* Adds the *Session ID* field to the *Highlighted fields* section on the Alert details flyout ({pull}132219[#132219]).
* Introduces a new design for the main navigation menu ({pull}132210[#132210], {pull}131437[#131437]).
* Adds Dashboards and Threat Hunting Landing pages ({pull}130905[#130905]).
* Introduces the Detections & Response dashboard ({pull}130670[#130670], {pull}129021[#129021], {pull}131029[#131029], {pull}128335[#128335], {pull}128087[#128087]).
** Update Detections Response view with pagination and opening numbers in Timeline ({pull}131828[#131828], {pull}131029[#131029]).
* Allows highlighted fields to be investigated in Timeline ({pull}131255[#131255]).
* Adds support to Osquery for multi-line queries ({pull}131224[#131224]).

> * Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]):

Reviewer 4: That looks good enough to me, thanks @nastasha-solomon 👍 

If I was nitpicking, I'd either formulate it around the bulk actions menu in the UI (so not mentioning the endpoint) or around the endpoint parameters.

If we'd emphasize the UI, it could be:

```
* Improves the performance of these bulk actions on rules ({pull}130924[#130924]):
** Add index patterns
** Delete index patterns
** Add tags
** Delete tags
** Apply Timeline template
```

If we'd emphasize the endpoint, it could be:

```
* Improves the performance of editing actions (`"action": "edit"`) supported by the `/api/detection_engine/rules/_bulk_action` endpoint ({pull}130924[#130924]). This covers the following actions available in the Rules table UI:
** Add index patterns
** Delete index patterns
** Add tags
** Delete tags
** Apply Timeline template
```

Note that we don't have "Set tags" and "Set index patterns" exposed as menu items in the UI.

> * Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]):

Author: Thanks, @banderror ! I noticed the difference between the UI and the endpoint as well and modeled the list against what was listed for the endpoint in [these docs](https://www.elastic.co/guide/en/security/master/bulk-actions-rules-api.html#bulk-edit-object-schema) because it looked closer to what was mentioned in the PR description of https://github.com/elastic/kibana/pull/130924. Do you have a preference on which we focus on for the release notes?

> * Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]):

Author: Ahh doc bug emerges! 

![doc-bug-emerges](https://user-images.githubusercontent.com/79124755/175073691-7390de18-25c3-4226-a043-b4add36075ab.gif)

Would you mind filing a doc bug for this in the `security-docs` repo @banderror so we have all the necessary deets?

> * Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]):

Reviewer 4: @nastasha-solomon will do. Or will just open a PR for that. It should be a tiny change.

> * Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]):

Author: You opening a PR will probably be faster right now. Feel free to tag me and any of the other doc team members on the PR as reviewers - thanks!

> * Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]):

Reviewer 4: https://github.com/elastic/security-docs/pull/2130

* Adds an “Authentications” tab to the User details flyout ({pull}129456[#129456]).
* Adds the ability to investigate Osquery results in Timeline ({pull}128596[#128596]).
* Allows multiple alerts to be added to a case ({pull}130958[#130958]).
* Adds the option to delete case comments multiple alerts to be added to a case ({pull}130254[#130254]).
* Provides an option to select a severity level for a case ({pull}131626[#131626]).
* Adds the *Alerts* tab to cases ({pull}131883[#131883]).
* Adds the *Average time to close* metric to the Cases page ({pull}131909[#131909]).

[discrete]
[[bug-fixes-8.3.0]]
==== Bug fixes and enhancements
* Separates array values in the Alerts table with commas ({pull}133297[#133297]).
* Exposes the EQL search configurations `event_category_field`, `tiebreaker_field`, and `timestamp_field` through the rules API and UI for Event Correlation rules ({pull}132247[#132247]).
* Adds the *Session ID* field to the *Highlighted fields* section on the Alert details flyout ({pull}132219[#132219]).
* Introduces a new design for the main navigation menu ({pull}132210[#132210], {pull}131437[#131437]).

> * Adds Dashboards and Threat Hunting Landing pages ({pull}130905[#130905]).

Author: Making a personal note to follow up on this after the release. What's interesting is that this PR appears in the "Features" section _only_ in the RN preview but then shows up in both the "Features" and "Bug fixes and enhancements" sections when you generate the output/asciidoc file. The PR is only labeled with the `release_note:feature` label, so I wouldn't expect it to be added to both sections when the asciidoc file is generated so this is definitely something to investigate. In the meantime, I'll remove it from the "Bug fixes and enhancements" section.

> * Introduces the Detections & Response dashboard ({pull}130670[#130670], {pull}129021[#129021], {pull}131029[#131029], {pull}128335[#128335], {pull}128087[#128087]).

Author: Will remove these from the bug fixes and enhancements section.

** Update Detections Response view with pagination and opening numbers in Timeline ({pull}131828[#131828], {pull}131029[#131029]).
* Allows highlighted fields to be investigated in Timeline ({pull}131255[#131255]).
* Adds support to Osquery for multi-line queries ({pull}131224[#131224]).
* Enforces the `All` global privilege setting on Osquery ({pull}130523[#130523]).
* Adds the *Run Osquery* option to the *More actions* menu (*...*) in the Alerts table ({pull}131790[#131790]).
* Improves the performance of these actions on the bulk rule actions endpoint ({pull}130924[#130924]).
** `add_tags`
** `delete_tags`
** `set_tags`
** `add_index`
** `delete_index`
** `set_index_patterns`

> ** `set_timeline_template`

Reviewer 4: I think these are not correct. These are the constants we have in the code:

```ts
export enum BulkActionEditType {
  'add_tags' = 'add_tags',
  'delete_tags' = 'delete_tags',
  'set_tags' = 'set_tags',
  'add_index_patterns' = 'add_index_patterns',
  'delete_index_patterns' = 'delete_index_patterns',
  'set_index_patterns' = 'set_index_patterns',
  'set_timeline' = 'set_timeline',
}
```

@nastasha-solomon so it should be:

```
** `add_index_patterns`
** `delete_index_patterns`
** `set_index_patterns`
** `set_timeline`
```

It also turns out that the docs are not correct, because they say `add_index` etc:
https://www.elastic.co/guide/en/security/current/bulk-actions-rules-api.html#_possible_bulkeditaction_object_values

@vitaliidm could you please fix the docs?

Reviewer 4: @nastasha-solomon I ended up fixing it in https://github.com/elastic/security-docs/pull/2130

> * Adds the *User risk* tab to the user details page ({pull}130256[#130256]).

Author: Hm, this was only ever labeled as a new feature (`release_note:feature`) so I'm not sure how it got duplicated in the output. Something to investigate after the release!

* Fixes a bug that caused the rule details page to crash when users opened a deleted or non-existent rule ({pull}133867[#133867]).
* Allows threshold alerts to be investigated in Timeline if filters are not provided ({pull}133733[#133733]).
* Prevents events from being added to cases from Timeline ({pull}133410[#133410]).

> * Fixes All Users and All Hosts page should reset the current page when sorting ({pull}133111[#133111]).

Reviewer 5: ```suggestion
* Fixes a bug that prevented the All Users and All Hosts pages from reseting after being sorted ({pull}133111[#133111]).
```
not 100% about the content here, but the sentence looks better to me this way

> * Removes the filter and investigate in Timeline options from the Elastic Agent status in Highlighted fields ({pull}132829[#132829], {pull}132586[#132586]).

Reviewer 5: ```suggestion
* Removes the Filter and Investigate in Timeline options from the Elastic Agent status in Highlighted fields ({pull}132829[#132829], {pull}132586[#132586]).
```
Should we bold "Filter" and "Investigate in Timeline" since they are clickables?

Author: These are the shorthand versions of the button names, figured they'd be easier to read and more succinct. The filter function in and out buttons aren't named in the UI and the add to timeline option is actually named **Add to timeline investigation**.

* Improves the copy of Timeline tooltips ({pull}132756[#132756]).
* Fixes a validation bug that occurred when users were building a rule exception and changed the exception statement’s operator ({pull}131989[#131989]).
* Adds a checkmark to the pagination selection on the *Exceptions lists* page ({pull}131979[#131979]).
* Re-adds the success message that displays when users export an exceptions list ({pull}131952[#131952]).
* Updates import toast logic ({pull}131873[#131873]).
* Ensures an error is not generated when the `agent.version` provided by an alert is in an unexpected format ({pull}131272[#131272]).
* Improves error checks for threshold rules ({pull}131088[#131088]).
* Expands support for migrating legacy rule actions ({pull}130511[#130511]).

> * Fixes a bug that caused the *Add Rule Exception* flyout to unexpectedly close when users create the first exception for the rule from an alert ({pull}130187[#130187]).

Reviewer 5: ```suggestion
* Fixes a bug that caused the *Add Rule Exception* flyout to unexpectedly close when users created the first exception for a rule from an alert ({pull}130187[#130187]).
```

> * Corrects Rule name sorting so detection rules are ordered alphabetically, regardless of their casing ({pull}130105[#130105]).

Reviewer 5: ```suggestion
* Corrects Rule name sorting so detection rules are ordered alphabetically, regardless of capitalization ({pull}130105[#130105]).
```

Reviewer: Revive's @benironside's suggestion, plus one additional edit:
```suggestion
* Corrects rule name sorting so detection rules are ordered alphabetically, regardless of capitalization ({pull}130105[#130105]).
```

* Improves the *Reporter* column in the Cases table ({pull}132200[#132200]).
* Adds the option to create a new case to the Select case pane ({pull}128882[#128882]).
* Allows {kibana-ref}/pre-configured-connectors.html[preconfigured connectors] to be used with cases ({pull}130372[#130372]).
* Inserts the deprecated icon next to deprecated preconfigured connectors ({pull}132237[#132237]).

> * Updates the Case table so that all tags assigned to the case are displayed when users go to the case and hover over the *Tags* column ({pull}132023[#132023).

Reviewer 5: ```suggestion
* Updates the Case table so that all tags assigned to a case are displayed when users go to the case and hover over the *Tags* column ({pull}132023[#132023).
```

* Adds Oauth support to the {sn} ITSM, SecOps, and ITOM connectors ({pull}131248[#131248).

> * Adds a setting to specify a list of allowed email domains which can be used with the email connector ({pull}129001[#129001).

Reviewer 5: ```suggestion
* Adds a setting that specifies the list of allowed email domains which can be used with the email connector ({pull}129001[#129001).
```
